server/app.js

```diff
@@ -0,0 +1,82 @@
+const express = require("express");
+const app = express();
+const stripe = require("stripe")(
+  "sk_test_51OqXAgSINGykgmo3WnHByaJLJ1QXMFZKoSZtWe4wWQCcBefP03qVG1Oywhif7HB0ayxqbhhqeZsYeJKUjAJ4iZ9u00ajtAcFBs"
+);
+const cors = require("cors");
+
+app.use(cors());
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+const COUPENCODE = [
+  {
+    Code: "ABC",
+    discountPercentage: 25,
+  },
+  {
+    Code: "BCD",
+    discountPercentage: 15,
+  },
+  {
+    Code: "CDE",
+    discountPercentage: 10,
+  },
+  {
+    Code: "XYZ",
+    discountPercentage: 5,
+  },
+];
+
+app.post("/api/v1/create-checkout-session", async (req, res) => {
+  const { products, promoCode } = req.body;
+
+  const CoupenDiscount = promoCode
+    ? COUPENCODE.find((item) => item.Code === promoCode)
+    : null;
+
+  const lineItems = products.map((product) => {
+    let unitPrice = product.item.price * product.quantity;
+    if (product.item.discountPercentage) {
+      unitPrice -= Math.round(
+        (unitPrice * product.item.discountPercentage) / 100
+      );
+    }
+    if (CoupenDiscount) {
+      unitPrice -= (unitPrice * CoupenDiscount.discountPercentage) / 100;
+    }
+    return {
+      price_data: {
+        currency: "usd",
+        product_data: {
+          name: product.item.title,
+          images: [product.item.images[0]],
+        },
+        unit_amount: Math.round(unitPrice * 100),
+      },
+      quantity: product.quantity,
+    };
+  });
+
+  try {
+    const session = await stripe.checkout.sessions.create({
+      payment_method_types: ["card"],
+      shipping_address_collection: {
+        allowed_countries: ["US", "CA", "GB", "AU"],
+      },
+      billing_address_collection: "auto",
+      line_items: lineItems,
+      mode: "payment",
+      success_url: "http://localhost:5173/success",
```

Comment:
save domains into env as well. So when code moves to different environment then you don't have to do code changes

Reply:
should i do same for frontend-side code where we are using Dummy.json API and and Chekout-session API in Cart.jsx.

Reply:
yes. If domain value is same then use one variable or define as per requirement.